invenio_rdm_records/config.py

```diff
@@ -421,3 +337,144 @@ def make_doi(prefix, record):
 This is only required if you want your records to be harvestable (OAI-PMH)
 in DataCite XML format.
 """
+
+
+"""Custom fields."""
+
+RDM_CUSTOM_FIELDS = []
+"""Records custom fields definition.
+
+Of the shape:
+
+.. code-block:: python
+
+    [
+        <custom-field-class-type(name='field')>,
+        # ...
+        <custom-field-class-type(name='fieldN')>'
+    ]
+
+For example:
+
+.. code-block:: python
+
+    [
+        TextCF(name="experiment"),
+        ...
+    ]
+"""
+
+
+RDM_CUSTOM_FIELDS_UI = []
+"""Upload form custom fields UI configuration.
+
+Of the shape:
+
+.. code-block:: python
+
+    [{
+        section: <section_name>,
+        fields: [
+            {
+                field: "path-to-field",  # this should be validated against the defined fields in `RDM_CUSTOM_FIELDS`
+                ui_widget: "<ui-widget-name>",  # predefined or user defined ui widget
+                props: {
+                    label:"<ui-label-to-display>",
+                    placeholder:"<placeholder-passed-to-widget>",
+                    icon:"<icon-passed-to-widget>",
+                    description:"<description-passed-to-widget>",
+                }
+            },
+        ],
+
+        # ...
+    }]
+
+For example:
+
+.. code-block:: python
+
+    [{
+        "section": "CERN Experiment"
+        "fields" : [{
+            field: "experiment",  # this should be validated against the defined fields in `RDM_CUSTOM_FIELDS`
+            ui_widget: "CustomTextField",  # user defined widget in my-site
+            props: {
+                label: "Experiment",
+                placeholder: "Type an experiment...",
+                icon: "pencil",
+                description: "You should fill this field with one of the experiments e.g LHC, ATLAS etc.",
+            }
+        },
+        ...
+    }]
+"""
+
+
+COMMUNITIES_CUSTOM_FIELDS = {}
```

Comment:
move this and below to `invenio-communities` and then it will be part of `cookiecutter`/`invenio.cfg`?

Reply:
I'm going to leave it as is, not to break it now. But there is an issue already to do this in the board.